components/typography/styles.js

```diff
@@ -41,26 +41,31 @@ export const bodyStandardStyles = css`
 	}
 `;
 
-export const bodyCompactStyles = css`
-	.d2l-body-compact {
-		font-size: 0.8rem;
-		font-weight: 400;
-		line-height: 1.2rem;
-	}
-	:host([skeleton]) .d2l-body-compact.d2l-skeletize::before {
-		bottom: 0.3rem;
-		top: 0.3rem;
-	}
-	:host([skeleton]) .d2l-body-compact.d2l-skeletize-paragraph-2 {
-		max-height: 2.4rem;
-	}
-	:host([skeleton]) .d2l-body-compact.d2l-skeletize-paragraph-3 {
-		max-height: 3.6rem;
-	}
-	:host([skeleton]) .d2l-body-compact.d2l-skeletize-paragraph-5 {
-		max-height: 6rem;
-	}
-`;
+export const generateBodyCompactStyles = (selector) => {
```

Comment:
As we discussed outside this review, I think I'd like to see this go through a `validateSelector()` step before being passed to `unsafeCSS`. It could use a simple regular expression to make sure the value is either a CSS class, id or tag name.

Reply:
Being explicit might be the safest.  If we need to support complex selectors including combinators (ex. `.foo .bar`), [this](https://stackoverflow.com/questions/12575845/what-is-the-regex-of-a-css-selector) might provide some ideas on what to look for.

Reply:
Just thinking a little more on the complex selector case.

Give this:
```css
:host([skeleton]) ${selector}.d2l-skeletize::before
```

This would be no good if the consumer tried to do:
```javascript
generateLabelStyles(':host .some-element')
```

Reply:
Good point -- agreed.